src/kOS.Safe/Encapsulation/QueueValue.cs

```diff
@@ -13,71 +13,59 @@ public QueueValue() : this(new Queue<T>())
         {
         }
 
-        public QueueValue(IEnumerable<T> queueValue) : base(new Queue<T>(queueValue))
+        public QueueValue(IEnumerable<T> queueValue) : base("QUEUE", new Queue<T>(queueValue))
         {
             QueueInitializeSuffixes();
         }
 
-        public override int Count
-        {
-            get { return Collection.Count; }
-        }
-
         public T Pop()
         {
-            return Collection.Dequeue();
+            return InnerEnumerable.Dequeue();
         }
 
         public void Push(T val)
         {
-            Collection.Enqueue(val);
+            InnerEnumerable.Enqueue(val);
         }
 
         public override Dump Dump()
         {
             var result = new DumpWithHeader
             {
-                Header = "QUEUE of " + Collection.Count() + " items:"
+                Header = "QUEUE of " + InnerEnumerable.Count() + " items:"
```

Comment:
Adding that method to Structure is a great idea.  If we make the change here, we'll want to make sure that it's handled throughout the serialization methods.